nodebuilder/share/share.go

```diff
@@ -1,38 +1,66 @@
-	"github.com/libp2p/go-libp2p-core/host"
-	"github.com/libp2p/go-libp2p-core/routing"
-	routingdisc "github.com/libp2p/go-libp2p/p2p/discovery/routing"
+	"github.com/ipfs/go-blockservice"
+	"go.uber.org/fx"
 
 	"github.com/celestiaorg/celestia-node/share"
-	"github.com/celestiaorg/celestia-node/share/availability/cache"
-	"github.com/celestiaorg/celestia-node/share/availability/discovery"
+	"github.com/celestiaorg/nmt/namespace"
 )
 
-func Discovery(cfg Config) func(routing.ContentRouting, host.Host) *discovery.Discovery {
-	return func(
-		r routing.ContentRouting,
-		h host.Host,
-	) *discovery.Discovery {
-		return discovery.NewDiscovery(
-			h,
-			routingdisc.NewRoutingDiscovery(r),
-			cfg.PeersLimit,
-			cfg.DiscoveryInterval,
-			cfg.AdvertiseInterval,
-		)
-	}
+// Module provides access to any data square or block share on the network.
+//
+// All Get methods provided on Module follow the following flow:
+//  1. Check local storage for the requested Share.
+//  2. If exists
+//     * Load from disk
+//     * Return
+//  3. If not
+//     * Find provider on the network
+//     * Fetch the Share from the provider
+//     * Store the Share
+//     * Return
+//
+// Any method signature changed here needs to also be changed in the API struct.
+//
+//go:generate mockgen -destination=mocks/api.go -package=mocks . Module
+type Module interface {
+	// SharesAvailable subjectively validates if Shares committed to the given Root are available on the Network.
+	SharesAvailable(context.Context, *share.Root) error
+	// ProbabilityOfAvailability calculates the probability of the data square
+	// being available based on the number of samples collected.
+	ProbabilityOfAvailability() float64
+	GetShare(ctx context.Context, dah *share.Root, row, col int) (share.Share, error)
+	GetShares(ctx context.Context, root *share.Root) ([][]share.Share, error)
+	// GetSharesByNamespace iterates over a square's row roots and accumulates the found shares in the given namespace.ID.
+	GetSharesByNamespace(ctx context.Context, root *share.Root, namespace namespace.ID) ([]share.Share, error)
 }
 
-// CacheAvailability wraps either Full or Light availability with a cache for result sampling.
-func CacheAvailability[A share.Availability](lc fx.Lifecycle, ds datastore.Batching, avail A) share.Availability {
-	ca := cache.NewShareAvailability(avail, ds)
+func NewModule(lc fx.Lifecycle, bServ blockservice.BlockService, avail share.Availability) Module {
+	serv := service.NewShareService(bServ, avail)
 	lc.Append(fx.Hook{
-		OnStop: ca.Close,
+		OnStart: func(ctx context.Context) error {
+			return serv.Start(ctx)
+		},
+		OnStop: func(ctx context.Context) error {
+			return serv.Stop(ctx)
+		},
 	})
-	return ca
+	return serv
+}
+
+// API is a wrapper around Module for the RPC.
+// TODO(@distractedm1nd): These structs need to be autogenerated.
+//
+//go:generate go run github.com/golang/mock/mockgen -destination=mocks/api.go -package=mocks . Module
```

Comment:
What's the meaning of this command? Seems like it generates mocks for testing, why is it in this comment

Reply:
https://go.dev/blog/generate

They are in every module